anilist.py

```diff
@@ -1,1172 +1,1182 @@
-                                (title_english, started_year, match_year))
-    if media_id is None:
-        logger.error('[ANILIST] No match found for title: %s' % (title))
-    return media_id
-
-
-def add_by_id(
-        anilist_id,
-        plex_title,
-        plex_year,
-        plex_watched_episode_count,
-        ignore_year):
-    matched_anilist_series = []
-    media_lookup_result = search_by_id(anilist_id)
-    if(media_lookup_result):
-        anilist_obj = search_item_to_obj(media_lookup_result)
-        if(anilist_obj):
-            matched_anilist_series.append(anilist_obj)
-            update_entry(
-                plex_title,
-                plex_year,
-                plex_watched_episode_count,
-                matched_anilist_series,
-                ignore_year)
-        else:
-            logger.error(
-                '[ANILIST] failed to get anilist object for list adding, skipping series')
-    else:
-        logger.error(
-            '[ANILIST] failed to get anilist search result for id: %s' %
-            (anilist_id))
-
-
-def update_series(mediaId, progress, status):
-    if ANILIST_SKIP_UPDATE == 'true':
-        logger.warning(
-            'Skip update is enabled in settings so not updating this item')
-        return
-    query = '''
-        mutation ($mediaId: Int, $status: MediaListStatus, $progress: Int) {
-            SaveMediaListEntry (mediaId: $mediaId, status: $status, progress: $progress) {
-                id
-                status,
-                progress
-            }
-        }
-        '''
-
-    variables = {
-        'mediaId': mediaId,
-        'status': status,
-        'progress': int(progress)
-    }
-
-    url = 'https://graphql.anilist.co'
-
-    headers = {
-        'Authorization': 'Bearer ' + ANILIST_ACCESS_TOKEN,
-        'Accept': 'application/json',
-        'Content-Type': 'application/json'
-    }
-
-    response = requests.post(
-        url, headers=headers, json={
-            'query': query, 'variables': variables})
-    # print(response.content)
-
-def retrieve_custom_mapping(title, season):
-    if custom_mappings:
-        for mapping in custom_mappings:
-            if mapping.title.lower().strip() == title.lower(
-            ).strip() and str(mapping.season) == str(season):
-                return mapping.anime_id
-
-    return 0
+import collections
+import inflect
+import json
+import logging
+import re
+import requests
+from guessit import guessit
+
+import plexmodule
+
+custom_mappings = []
+ANILIST_ACCESS_TOKEN = ''
+ANILIST_SKIP_UPDATE = 'false'
+ANILIST_PLEX_EPISODE_COUNT_PRIORITY = 'false'
+
+# Set this to True for logging failed AniList matches to failed_matches.txt file
+ANILIST_LOG_FAILED_MATCHES = False
+
+
+def to_object(o):
+    keys, values = zip(*o.items())
+    # print(keys, values)
+    return collections.namedtuple('X', keys)(*values)
+
+
+def int_to_roman_numeral(insert):

```

Comment:
Shouldn't this be `integer` instead of `insert`?

Reply:
Yea I think it should no idea how it got there, weird enough that the code still works but gonna update it soon and check it again, writing the README.md right now.